agents/background/worker.py

```
@@ -0,0 +1,325 @@
+    
+    Returns:
+        bool: `True` if the request is allowed, `False` if the rate limit has been exceeded.
+    """
+    if not redis_conn:
+        logger.critical("Redis connection missing. Failing open for safety.")
+        return True
+
+
+    key = f"rate_limit:{client_id}"
+    try:
+        if current_count == 1 or redis_conn.ttl(key) == -1:
```

Ruff (0.14.10)

Undefined name `current_count`

(F821)